### PR TITLE
WordPress CMS Integration (Part 1)

### PR DESCRIPTION
**Pull request by maxharrisnet**
This starts our process of connecting the frontend of the site to the CMS. The magic is in `js/cms.js` where we have a few functions that call the WordPress API to get page and event information. 

So far in WordPress, we have the fields for Events and some common fields for pages (see screenshots). 
Screen Shot 2024-01-24 at 11 02 33 AM
Screen Shot 2024-01-24 at 11 02 18 AM

With these fields I was able to populate the following pages in WordPress:
- About Us
- Contact Us
- Privacy Policy
- Terms of Service
- Accessibility

In this PR, we updating these pages to fetch that data and populate the frontend (only if the data from WP exists). 

To test this, you can open the page and view the content. I prefixed all of the content with ### in WordPress so that we can easily differentiate it from the static content.

Next steps are to create the unique page fields in WordPress and link those page files to the CMS. This is a good starting place, so I will include those changes in a subsequent PR. 

I hope this all makes sense. @SolivanLau please let me know if anything needs clarification.

### CODE REVIEW

File: about.html [hunk 1]
```diff
@@ -1,306 +1,295 @@
...
+					<a href="/contact.html" class="btn-main desktop-support-btn">Contact Us</a>
 				</div>
+			</div>
+		</nav>
+		<!-- Hero -->
+		<section class="hero">
+			<img
+				src="/assets/images/pages/about/hero-about.jpg"
+				alt="homepage background"
+				class="img-mobile" id="hero-image" />
```
Comment (maxharrisnet) [inline about.html:88]: Adding these ids (`hero-image`, `hero-heading`, `hero-subheading`) to link them to the CMS in JavaScript

File: about.html [hunk 2]
```diff
@@ -1,306 +1,295 @@
...
+						nations, and communities as we engage in our work.
+					</p>
+				</div>
+			</div>
+		</div>
+	</div>
+
+	<!-- JavaScript -->
+	<script src="/js/main.js"></script>
+	<script src="https://cdn.jsdelivr.net/npm/marked/marked.min.js"></script>
```
Comment (maxharrisnet) [inline about.html:288]: Marked helps us parse the data from the WYSIWYG editor into HTML. This is for the content sections.

File: about.html [hunk 3]
```diff
@@ -1,306 +1,295 @@
...
+	<script src="/js/cms.js"></script>
+	<script>
+		getPageData('about');
```
Comment (maxharrisnet) [inline about.html:291]: This uses the page slug in WordPress 

Screen Shot 2024-01-24 at 11 17 29 AM

File: js/cms.js [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
-function getPageData(page, source) {
-	const cms = source;
-	let apiURL = "";
-
-	if (cms === "strapi") {
-		apiURL = `https://cms.iftheselandscouldtalk.org/api/${page}?populate=*`;
-	} else if (cms === "wordpress") {
-		apiURL = `https://wp.iftheselandscouldtalk.org/wp-json/wp/v2/page`;
+// Helper functions
+function appendData(data, element, method, target) {
```
Comment (maxharrisnet) [inline js/cms.js]: This is the main function appending the data to the various elements and containers. It works, but it's a little brittle right now. I need to add some checks to make sure the targets exist, etc.

Comment (SolivanLau) [reply js/cms.js]: Looks good. Lots of checks for type of data.

File: js/cms.js [hunk 5]
```diff
@@ -1,13 +1,30 @@
...
-	} else if (cms === "wordpress") {
-		apiURL = `https://wp.iftheselandscouldtalk.org/wp-json/wp/v2/page`;
+// Helper functions for appending data to DOM elements
+function appendData(data, element, method, target) {
+	if (data) {
+		const el = document.createElement(element);
+
+		if (method === "text") {
+			target.textContent = "";
+			target.appendChild(document.createTextNode(data));
```
Comment (SolivanLau) [inline js/cms.js]: looks  versatile currently. Seems like major types of content can dynamically appended  to specified target element based on the method and input data. might be worth sharing and walking through the if  conditional statements w the class

File: js/cms.js [hunk 6]
```diff
@@ -88,43 +102,43 @@ function getCollectionData(collection, source) {
...
+				card.classList.add("card", "event");
+				card.href = `events/event.html?e=${post.slug}`;
+
+				// Append data to the card
+				if (data.featured_image.url) {
+					const image = document.createElement("img");
+					image.src = data.featured_image.url;
+					data.featured_image.alt
+						? (image.alt = data.featured_image.alt)
+						: "If These Lands Could Talk";
```
Comment (SolivanLau) [inline js/cms.js:143]: nice alternative with the ternary operator. It could also be the **.title** of the post if there is no **.alt** content from cms

**Review by SolivanLau — APPROVED** [review]
Hello, the cms work looks good. No changes but just a question on my end.

- could avoid the if else if else in the append function if order does not necessarily matter? maybe having independent if statements  could make code more readable and consistent w the other if statements in other cms functionality.
- left a small note of using the **.title** as  an alternative to **.alt** if the image does not have an alt value

File: js/cms.js (hunk 6 above)
Comment (maxharrisnet) [reply js/cms.js:143]: Yes, good call

**Pull request merged**